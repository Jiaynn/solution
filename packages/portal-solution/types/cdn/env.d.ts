/**
 * @file env config
 * @author linchen <linchen@qiniu.com>
 */

declare interface OEMConfig {
  logo: string
  vendor: number
  favicon: string
  title: string
  copyright: string
  lang: 'en' | 'cn'
  domainHosting: boolean                  // 域名托管功能
  financial: boolean                      // 财务中心
  hideLoginTitle: boolean                 // 隐藏登录页面的 title
  hideCreateDomain: boolean               // 隐藏创建域名的操作入口
  hideUpdateDomain: boolean               // 隐藏更新域名的操作入口
  hideSubAccountDomainListCname: boolean  // 对子用户隐藏域名列表页的 cname
  hideSubAccountDomainDetailCname: boolean // 对子用户隐藏域名配置详情页的 cname
  hideSubAccountDomainOperations: boolean // 对子用户隐藏域名的操作 (只保留统计操作)
  hideSubAccountCreateDomain: boolean     // 对子用户隐藏创建域名的操作
  hideSubAccountGeoForeign: boolean       // 对子用户禁用海外覆盖范围
  hideSubAccountGeoGlobal: boolean        // 对子用户禁用全球覆盖范围
  hideSubAccountStatsUsagePeakAverage: boolean    // 对子用户隐藏平均峰值带宽
  hideSubAccountStatsUsagePeak95Average: boolean  // 对子用户隐藏平均 95 峰值带宽
  statisticsDateRangeDays: number // 数据统计默认时间范围, 最近的天数
  statisticsFreq: string // 数据统计粒度 '1day' | '1hour' | '5min'
}

declare let BUILD_TARGET: 'qiniu' | 'oem'
declare let OEM_CONFIG: OEMConfig
declare let APP_VERSION: string
