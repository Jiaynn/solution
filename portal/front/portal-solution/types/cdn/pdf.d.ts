declare let pdfExport: string

declare module '*.pdf' {
  export = pdfExport
}
