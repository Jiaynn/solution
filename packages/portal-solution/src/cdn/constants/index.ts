// 启用动态加速工单地址
export const enableDynamicURL = 'https://support.qiniu.com/tickets/new/form?category=%E5%85%B6%E4%BB%96%E7%B1%BB%E5%92%A8%E8%AF%A2&space=%E8%9E%8D%E5%90%88%20CDN'

// 证书配置提交工单 URL
export const certConfigHelpURL = 'https://support.qiniu.com/tickets/new/form?category=%E8%AF%81%E4%B9%A6%E9%97%AE%E9%A2%98&space=CDN'
