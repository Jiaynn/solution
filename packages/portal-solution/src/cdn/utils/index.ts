import { values } from 'lodash'
import { ReactNode } from 'react'
import { makeCancelled, UnknownException } from 'qn-fe-core/exception'

export function isPromise(obj: any): obj is Promise<any> {
  if (!obj) {
    return false
  }
  if (typeof obj === 'object' || typeof obj === 'function') {
    if (typeof obj.then === 'function') {
      return true
    }
  }
  return false
}

export interface IAntdStyleMethodOptions {
  title?: ReactNode
  content?: ReactNode
  onOk?: (...args: any[]) => void
  onCancel?: (...args: any[]) => void
}

export type AntdStyleMethod<T extends IAntdStyleMethodOptions> = (options: T) => void

export type PromiseStyleMethod<T, P> = (options: T) => Promise<P>

// 把 antd 中如 Modal.confirm 风格（onCancel, onOk）的接口转换成 promise 接口
export function antdToPromise<
  T extends IAntdStyleMethodOptions,
  M extends AntdStyleMethod<T>,
  P
>(method: M): PromiseStyleMethod<T, P> {
  return (options: T) => new Promise((resolve, reject) => {
    const onCancel = (...args: any[]) => {
      if (options.onCancel) {
        options.onCancel(...args)
      }
      reject(makeCancelled())
    }
    const onOk = (...args: any[]) => {
      if (options.onOk) {
        options.onOk(...args)
      }
      resolve(args[0])
    }
    method({
      ...(options as any),
      onCancel,
      onOk
    })
  })
}

function getLexicValue(char: string) {
  return char ? char.charCodeAt(0) : -1
}

export function lexicCompare(strA: string, strB: string) {
  for (let i = 0, len = Math.max(strA.length, strB.length); i < len; i++) {
    const vA = getLexicValue(strA.charAt(i))
    const vB = getLexicValue(strB.charAt(i))
    if (vA !== vB) {
      return vA - vB
    }
  }
  return 0
}

export function valuesOfEnum<T extends object>(target: T): Array<T[keyof T]> {
  const fakeKeys = values(target).filter(value => typeof value !== 'string').map(String)
  const realKeys = Object.keys(target).filter(key => fakeKeys.indexOf(key) === -1)

  return (realKeys as Array<keyof T>).map(key => target[key])
}

export function assertUnreachable(): never
export function assertUnreachable(v: never, message?: string): never
export function assertUnreachable(v?: never, message = "Didn't expect to get here") {
  throw new UnknownException(message + v)
}

/**
 * 过滤数组的 “空” 值，其中空值指的是：
 * - null
 * - undefined
 */
export function nonEmptyArray<T>(data: Array<T | undefined | null>): T[] {
  return data.filter((it): it is T => it != null)
}

export function booleanPredicate<T>(value: T): value is Exclude<T, false | null | undefined | '' | 0> {
  return Boolean(value)
}

export function assert(condition: any, msg?: string): asserts condition {
  if (!condition) {
    throw Error(msg)
  }
}
