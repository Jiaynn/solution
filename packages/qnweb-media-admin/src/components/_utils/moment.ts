import moment from 'moment';

/**
 * 格式化duration
 * @param duration 时长，单位毫秒(ms)
 */
export const formatDuration = (duration?: number) => {
  return moment.utc(duration).format('HH:mm:ss');
};

/**
 * 格式化时间
 * @param datetime
 * @param format
 */
export const formatDatetime = (datetime: moment.MomentInput, format?: string) => {
  return moment(datetime).format(format || 'YYYY-MM-DD HH:mm:ss');
};

/**
 * 毫秒=>秒
 * @param value
 */
export const millisecondToSecond = (value: number) => {
  return value / 1000;
};

/**
 * 秒=>毫秒
 * @param value
 */
export const secondToMillisecond = (value: number) => {
  return value * 1000;
};
