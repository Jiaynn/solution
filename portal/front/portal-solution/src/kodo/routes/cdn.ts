/**
 * @file CDN route TODO: 站内外链与内链分开
 * @author yinxulai <me@yinxulai.com>
 */

// CDN 概览
export function getCDNOverviewPath() {
  return '/cdn/overview'
}

// CDN 业务的域名创建
export function getCDNCreateDomainPath(bucketName: string) {
  const query = bucketName ? `?bucket=${encodeURIComponent(bucketName)}&fixBucket` : ''
  return `/cdn/domain/create${query}`
}

// CDN 业务的域名列表
export function getCDNDomainListPath() {
  return '/cdn/domain'
}

export function getCDNDomainDetailPath(domain: string) {
  return `/cdn/domain/${domain}`
}

export function getCDNCreateBucketDomainPath(bucketName: string) {
  return `/cdn/domain/create?bucket=${bucketName}&fixBucket`
}

export function getCDNRefreshPath() {
  return '/cdn/refresh-prefetch'
}

export function getCDNRefreshLogsPath() {
  return '/cdn/refresh-prefetch?tab=showLogs'
}
