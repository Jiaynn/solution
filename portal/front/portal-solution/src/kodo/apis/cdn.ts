/**
 * @file config api
 * @description region 相关的接口
 * @author lizhifeng <lizhifeng@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { injectable } from 'qn-fe-core/di'
import { ApiException } from 'qn-fe-core/client'

import { ProxyClientV2 } from 'portal-base/common/apis/proxy'

import { proxy } from 'kodo/constants/apis'

interface CdnBaseResponse {
  code: number
  error: string
  [k: string]: any
}

export interface RefreshCdnSurplus extends CdnBaseResponse {
  urlQuotaDay: number // url 刷新当日限额
  urlSurplusDay: number // url 刷新当日余量
  dirQuotaDay: number // 目录刷新当日限额
  dirSurplusDay: number // 目录刷新当日余量
  quotaDay: number // 预取当日限额
  surplusDay: number // 预取当日剩余量
}

interface RefreshCdnOptions {
  product: 'cdn'
  urls: string[]
}

export class CdnApiException extends ApiException {
  constructor(
    public code: number,
    public payload: CdnBaseResponse,
    message?: string,
    cause?: unknown
  ) {
    super(
      'RefreshPrefetchApiException',
      code,
      message,
      undefined,
      cause
    )
  }
}

const messageMap = new Map<number, string>([
  [400031, '无效的 URL'],
  [400034, '刷新域名个数超过限制'],
  [400037, 'URL 已在队列中，无需重复提交'],
  [400039, '请求次数过多']
])

function produceResult<T extends CdnBaseResponse>(payload: T) {
  if (payload && typeof payload === 'object' && 'code' in payload && payload.code !== 200) {
    const message = messageMap.get(payload.code) || payload.error
    throw new CdnApiException(payload.code, payload, message)
  }

  return payload
}

@autobind
@injectable()
export class CdnApis {
  constructor(
    private proxyClient: ProxyClientV2
  ) {}

  getRefreshCdnSurplus(): Promise<RefreshCdnSurplus> {
    return this.proxyClient.get<RefreshCdnSurplus>(proxy.refreshCdnSurplus, {})
      .then(produceResult)
  }

  async refreshCdn(params: RefreshCdnOptions): Promise<void> {
    await this.proxyClient.post<CdnBaseResponse>(proxy.refreshCdn, params)
      .then(produceResult)
  }
}
