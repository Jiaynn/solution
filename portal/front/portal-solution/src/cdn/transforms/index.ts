import { QueryParams } from 'qn-fe-core/utils'

export function atoi(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseInt(numberText, 10)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function atof(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseFloat(numberText)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function getIntValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atoi((e.target as any).value, valueForInvalidInput)
}

export function getFloatValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atof((e.target as any).value, valueForInvalidInput)
}

export function trimAndFilter(lines: string[]): string[] {
  return (lines || []).map(
    line => line.trim()
  ).filter(
    line => !!line
  )
}

export function splitLines(linesStr: string | null | undefined): string[] {
  if (!linesStr) {
    return []
  }
  return linesStr.split('\n')
}

export function filterQuery(query?: QueryParams) {
  if (!query) {
    return
  }
  return Object.keys(query).reduce(
    (filtered, key) => (
      (query || {})[key]
      ? { ...filtered, [key]: (query || {})[key] }
      : filtered
    ), {} as QueryParams
  )
}

export function humanizeOnoff(isOn: boolean) {
  return isOn ? '开启' : '关闭'
}
