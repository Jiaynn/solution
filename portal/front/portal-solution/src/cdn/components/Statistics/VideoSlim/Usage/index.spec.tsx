import React from "react";
import moment from "moment";

import { RendererUtils as Renderer } from "cdn/test";

jest.mock("react-icecream-charts/lib/area", () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  AreaChart: (props: React.PropsWithChildren<{}>) => (
    <p>{JSON.stringify(props, null, 2)}</p>
  ),
}));

jest.mock("antd/lib/spin", () => ({
  __esModule: true,
  default: (props: React.PropsWithChildren<{}>) => (
    <div className="mock-antd-spin">{props.children || null}</div>
  ),
}));
import VideoSlimUsage from ".";

it("renders correctly", () => {
  const renderer = new Renderer();
  const tree = renderer
    .createWithAct(
      <VideoSlimUsage
        options={{
          domains: ["foo.com"],
          startDate: moment("2018-12-01"),
          endDate: moment("2018-12-04"),
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
