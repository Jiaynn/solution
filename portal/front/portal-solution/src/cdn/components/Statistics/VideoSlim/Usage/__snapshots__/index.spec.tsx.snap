// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="mock-antd-spin"
  >
    <div
      className="ant-row display-control"
    >
      <div
        className="ant-col ant-col-16"
      >
        <div>
          <div
            className="ant-col ant-col-6"
          >
            <div
              className="comp-statistics-summary-item"
            >
              <div
                className="summary-item-title"
              >
                瘦身总次数
              </div>
              <div
                className="summary-item-value"
              >
                --
              </div>
            </div>
          </div>
          <div
            className="ant-col ant-col-6"
          >
            <div
              className="comp-statistics-summary-item"
            >
              <div
                className="summary-item-title"
              >
                2K 视频总时长
              </div>
              <div
                className="summary-item-value"
              >
                --
              </div>
            </div>
          </div>
          <div
            className="ant-col ant-col-6"
          >
            <div
              className="comp-statistics-summary-item"
            >
              <div
                className="summary-item-title"
              >
                HD 视频总时长
              </div>
              <div
                className="summary-item-value"
              >
                --
              </div>
            </div>
          </div>
          <div
            className="ant-col ant-col-6"
          >
            <div
              className="comp-statistics-summary-item"
            >
              <div
                className="summary-item-title"
              >
                SD 视频总时长
              </div>
              <div
                className="summary-item-value"
              >
                --
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col ant-col-4 ant-col-offset-4"
      >
        <button
          className="ant-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <i
            aria-label="icon: download"
            className="anticon anticon-download"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="download"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M505.7 661a8 8 0 0 0 12.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </i>
          <span>
            导出 CSV
          </span>
        </button>
      </div>
    </div>
    <div
      className="chart"
    >
      <p>
        {
  "series": [],
  "chartOptions": {
    "tooltip": {
      "shared": true,
      "xDateFormat": "%Y-%m-%d %H:%M"
    },
    "xAxis": {
      "type": "datetime"
    },
    "yAxis": {
      "decimals": 0
    },
    "height": 300,
    "unit": [
      {
        "threshold": 0,
        "name": "次"
      },
      {
        "threshold": 1000,
        "name": "千次"
      }
    ],
    "decimals": 3
  }
}
      </p>
      <p
        className="chart-tips"
      >
        注：统计的视频时长，以秒四舍五入换算为分钟，与账单计费时长略有误差
      </p>
    </div>
  </div>
  <div>
    <div
      className="table-title-wrapper"
    >
      <h3
        className="title"
      >
        瘦身服务使用详情
      </h3>
      <div
        className="table-search-input"
      >
        <span
          className="ant-input-search ant-input-affix-wrapper"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-suffix"
          >
            <i
              aria-label="icon: search"
              className="anticon anticon-search ant-input-search-icon"
              onClick={[Function]}
              tabIndex={-1}
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </i>
          </span>
        </span>
      </div>
    </div>
    <div
      className="ant-table-wrapper"
    >
      <div
        className="mock-antd-spin"
      >
        <div
          className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              onWheel={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            #
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            时间
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            资源
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            文件格式
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            瘦身前后文件大小(MB)
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            瘦身后规格
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div>
                          <span
                            className="ant-table-column-title"
                          >
                            视频时长
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          />
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                />
              </table>
            </div>
            <div
              className="ant-table-placeholder"
            >
              <div
                className="ant-empty ant-empty-normal"
              >
                <div
                  className="ant-empty-image"
                >
                  <img
                    alt="No Data"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                  />
                </div>
                <p
                  className="ant-empty-description"
                >
                  No Data
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
